sw/airborne/modules/datalink/telemetry.c

```diff
@@ -48,9 +48,9 @@ struct periodic_telemetry pprz_telemetry = { TELEMETRY_PPRZ_NB_MSG, telemetry_cb
  * @param _cb callback function, called according to telemetry mode and specified period
  * @return -1 on failure to register, index of callback otherwise
  */
-int8_t register_periodic_telemetry(struct periodic_telemetry *_pt, uint8_t _id, telemetry_cb _cb)
```

Comment:
The input id can be extended to uint16, but the return type corresponds to the index in the registration table. So unless you plan to register more than 127 different messages in your telemetry file, it is not necessary to change its type.